src/components/common/Chatbox.tsx

```diff
@@ -1,38 +1,47 @@
-import { Record } from 'immutable';
-import PropTypes from 'prop-types';
-
-import { useContext } from 'react';
-
-import GraaspChatbox from '@graasp/chatbox';
+import GraaspChatbox, {
+  DeleteMessageFunctionParamType,
+  EditMessageFunctionParamType,
+  SendMessageFunctionParamType,
+} from '@graasp/chatbox';
 import { MUTATION_KEYS } from '@graasp/query-client';
 import { PermissionLevel } from '@graasp/sdk';
+import { ItemRecord } from '@graasp/sdk/frontend';
 import { Loader } from '@graasp/ui';
 
 import { hooks, useMutation } from '../../config/queryClient';
 import { CHATBOX_ID, CHATBOX_INPUT_BOX_ID } from '../../config/selectors';
-import { CurrentUserContext } from '../context/CurrentUserContext';
+import { useCurrentUserContext } from '../context/CurrentUserContext';
 
 const { useItemChat, useMembers, useAvatar, useItemMemberships } = hooks;
 
-const Chatbox = ({ item }) => {
+type Props = {
+  item: ItemRecord;
+};
+
+const Chatbox = ({ item }: Props): JSX.Element => {
   const { data: chat, isLoading: isChatLoading } = useItemChat(item.id);
   const { data: itemPermissions, isLoading: isLoadingItemPermissions } =
     useItemMemberships(item.id);
   const { data: members, isLoading: isMembersLoading } = useMembers(
     itemPermissions?.map((m) => m.memberId)?.toArray() || [],
   );
   const { data: currentMember, isLoading: isLoadingCurrentMember } =
-    useContext(CurrentUserContext);
-  const { mutate: sendMessage } = useMutation(
-    MUTATION_KEYS.POST_ITEM_CHAT_MESSAGE,
-  );
-  const { mutate: editMessage } = useMutation(
-    MUTATION_KEYS.PATCH_ITEM_CHAT_MESSAGE,
-  );
-  const { mutate: deleteMessage } = useMutation(
-    MUTATION_KEYS.DELETE_ITEM_CHAT_MESSAGE,
-  );
-  const { mutate: clearChat } = useMutation(MUTATION_KEYS.CLEAR_ITEM_CHAT);
+    useCurrentUserContext();
+  const { mutate: sendMessage } = useMutation<
+    unknown,
+    unknown,
+    SendMessageFunctionParamType
```

Comment:
Yes, these need to be typed when using the mutation. An improvement is to direclty export a typed funciton that specifies these types alongside the keys. I think @pyphilia is working on this right now. the `unknown` s are necesary because we need to type the third type which is the `TData`. there is no way around that.